host/windows/syscall.c

```diff
@@ -324,6 +327,475 @@ __declspec(noreturn) static void _panic(
+    // of really large numbers of disks (>26). If you have those, mount on
+    // windows.
+    //
+
+    int origin_len =  strnlen_s(path, MAX_PATH);
+    if (origin_len >= 2 && isalpha(path[0]) && path[1] == ':')
+    {
+        // Abosolute path, just replace c: to /c
+        required_size = origin_len + 1;
+
+        enclave_path = (char*)calloc(required_size, sizeof(char));
+        if (!enclave_path)
+        {
+            _set_errno(OE_ENOMEM);
+            goto done;
+        }
+        if (oe_memcpy_s(enclave_path, sizeof(char) * required_size, path,
+                    origin_len) != OE_OK)
+        {
+            _set_errno(OE_EINVAL);
+            goto done;
+        }
+    }
+    else
+    {
+        // Relative path, ./tmp or /tmp.
+        // /tmp means D:\tmp if pwd is under D:\.
+        //  \tmp is the same case.
+        // Anyway we need pwd.
+        current_dir = _getcwd(NULL, 0);
+        current_dir_len = strnlen_s(current_dir, MAX_PATH);
+
+        if (current_dir_len < 2 || !isalpha(current_dir[0]) || current_dir[1] != ':')
+        {
+            //_getcwd result is wrong
+            _set_errno(OE_EINVAL);
+            goto done;
+        }
+
+        if (path[0] == '\\' || path[0] == '/')
+        {
+            // Only need the drive name.
+            current_dir_len = 2;
+        }
+
+        required_size = current_dir_len + origin_len + 1;
+
+        enclave_path = (char*)calloc(required_size, sizeof(char));
+        if (!enclave_path)
+        {
+            _set_errno(OE_ENOMEM);
+            goto done;
+        }
+
+        if (oe_memcpy_s(enclave_path, sizeof(char) * required_size,
+                    current_dir, sizeof(char) * current_dir_len) != OE_OK)
+        {
+            _set_errno(OE_EINVAL);
+            goto done;
+        }
+        if (oe_memcpy_s(enclave_path + sizeof(char) * current_dir_len,
+                    sizeof(char) * (required_size - current_dir_len), path,
+                    sizeof(char) * origin_len) != OE_OK)
+        {
+            _set_errno(OE_EINVAL);
+            goto done;
+        }
+    }
+
+    // Clean up
+
+    // There are at least 2 chars are copied to enclave_path as disk:
+    // Check the length here and replace disk: at the first 2 position as /disk.
+    if (required_size >= 3 && isalpha(enclave_path[0]) && enclave_path[1] == ':')
+    {
+        enclave_path[1] = enclave_path[0];
+        enclave_path[0] = '/';
+    }
+    else
+    {
+        _set_errno(OE_EINVAL);
+        goto done;
+    }
+
+    enclave_path[required_size - 1] = '\0';
+    normalize_path(enclave_path, required_size, '/');
+
+done:
+    if (current_dir)
+    {
+        free(current_dir);
+    }
+    return enclave_path;
+}
+
+// Allocates WCHAR* string which follows the expected rules for
+// enclaves comminication with the host file system API. Paths in the format
+// /<driveletter>/<item>/<item>  become <driveletter>:\\<item>\<item>
```

Comment:
I think the best option is to sandbox the enclave's view of the file system. So on linux my host root might be `/tmp/oe12345` or `C:\Users\foo\Temp\oe12345`. Then from inside the enclave I just do `mkdir("/test")` and it could create `/tmp/oe12345/test` or `C:\Users\foo\Temp\oe12345\test`.

And then for specifying the root we make the application tell us the literal filepath from the OS with the correct slashes, drive letter, etc.